### PR TITLE
feat(rbac): ensure the builtin roles on auth

### PR DESCRIPTION
**Pull request by flaneur2020**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

When we have a multi-tenant management mode instance, it might have some corner cases when an user did not executed any query instance, but try querying user's connection info via the management mode instance. at this time, ACCOUNT_ADMIN might not get initialized.

This PR made these changes:

- added the initialization builtin-role logic after the authentication phase.
- sync the privileges to account_admin when any new privilege type get introduced.
- on the sqlogictest side, moved the initialization builtin-role logic to SettingInterpreteter instead of the low-level setting module.

## Tests

- [x] Logic Test: this PR keeps the logictest about roles not changed.

## Type of change

- [x] Bug Fix (non-breaking change which fixes an issue)

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/14451)

### CODE REVIEW

File: src/query/users/src/role_mgr.rs [hunk 1]
```diff
@@ -52,6 +67,55 @@ impl UserApiProvider {
...
 
+    /// Ensure the builtin roles account_admin and public exists. Please note that there's
+    /// a corner case that if we added another privilege type, we should add it to the
+    /// existed account_admin role.
+    ///
+    /// This function have two calling places:
+    /// 1. when the server starts
+    /// 2. when the user is authenticated
+    #[async_backtrace::framed]
+    pub async fn ensure_builtin_roles(&self, tenant: &str) -> Result<()> {
```
Comment (TCeason) [inline src/query/users/src/role_mgr.rs:78]: maybe we can add a ut about this? revoke a exists privilege in account_admin like SUPER. then call this function, the account_admin's privilege will contain SUPER

File: src/query/service/src/auth.rs [hunk 2]
```diff
@@ -161,9 +164,13 @@ impl AuthMgr {
 
                 authed?;
 
+                UserApiProvider::instance()
```
[the remark this answers is not in the document]
Comment (flaneur2020) [reply src/query/service/src/auth.rs:167]: IMO these not make quite differences, what we need take care is to put this line after authentication passed.

File: src/query/users/src/role_mgr.rs (hunk 1 above)
Comment (flaneur2020) [reply src/query/users/src/role_mgr.rs:78]: makes sense, let me add it

Comment (flaneur2020) [reply src/query/users/src/role_mgr.rs:81]: i've added a ut about this, PTAL

**Review by TCeason — APPROVED** [review]
LGTM

File: src/query/service/src/auth.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (TCeason) [reply src/query/service/src/auth.rs:167]: Ok

[the remark this answers is not in the document]
Comment (TCeason) [reply src/query/service/src/auth.rs:167]: I'm more worried about missing the ensure operation by adding other authentication logic. But for now it's fine

**Comment by TCeason** [comment]
I think we can merge this pr. Cc @BohuTANG @flaneur2020

**Pull request merged**